src/DroneSafetyApi/Services/Heatmap.cs

```diff
@@ -8,28 +9,55 @@ namespace DroneSafetyApi.Services
     public class HeatMap
     {
         private Dictionary<Position, int> Map;
-        public double StartX { get; set; }
-        public double StartY { get; set; }
-        public double EndX { get; set; }
-        public double EndY { get; set; }
-        public double Delta { get; set; }
+        private double StartX;
+        private double StartY;
+        private double EndX;
+        private double EndY;
+        private double DeltaX;
+        private double DeltaY;
 
-        public HeatMap(double minX, double maxX, double minY, double maxY, double resolution)
+        private int DecimalPlacesX;
+        private int DecimalPlacesY;
+
+        const int MetresInLatDegree = 110575;
+
+        public HeatMap(double minX, double maxX, double minY, double maxY, int width, int height)
         {
             StartX = minX;
             StartY = minY;
             EndX = maxX;
             EndY = maxY;
 
-            Delta = resolution;
+            DeltaX = (maxX - minX) / width;
+            DeltaY = (maxY - minY) / height;
+            DecimalPlacesX = 0;
+            while ((int)DeltaX % 10 == 0)
+            {
+                DecimalPlacesX++;
+                DeltaX *= 10;
+            }
+            DeltaX /= Math.Pow(10, DecimalPlacesX);
+            DecimalPlacesX += (int)Math.Log10(width);
+            DeltaX = Math.Round(DeltaX, DecimalPlacesX);
+
+            DecimalPlacesY = 0;
+            while ((int)DeltaY % 10 == 0)
+            {
+                DecimalPlacesY++;
+                DeltaY *= 10;
+            }
+            DeltaY /= Math.Pow(10, DecimalPlacesY);
+            DecimalPlacesY += (int)Math.Log10(height);
+            DeltaY = Math.Round(DeltaY, DecimalPlacesY);
 
             Map = new Dictionary<Position, int>();
             
         }
 
-        public void AddHazard(double x, double y, int v)
+        private void AddHazard(double x, double y, int v)
```

Comment:
What do you mean? This is in the heatmap class, and so are all the processing algorithms as you said they should be, so anything that should be added to the heatmap will be in the heatmap class, and so it should be private